ee/agent/storage/bbolt/keyvalue_store_bbolt.go

```diff
@@ -225,3 +228,66 @@ func (s *bboltKeyValueStore) Update(kvPairs map[string]string) ([]string, error)
 
 	return deletedKeys, nil
 }
+
+func (s *bboltKeyValueStore) Count() int {
+	if s == nil || s.db == nil {
+		s.slogger.Log(context.TODO(), slog.LevelError, "unable to count uninitialized bbolt storage db")
+		return 0
+	}
+
+	var len int
+	if err := s.db.View(func(tx *bbolt.Tx) error {
+		b := tx.Bucket([]byte(s.bucketName))
+		if b == nil {
+			return NewNoBucketError(s.bucketName)
+		}
+
+		len = b.Stats().KeyN
+		return nil
+	}); err != nil {
+		s.slogger.Log(context.TODO(), slog.LevelError,
+			"err counting from bucket",
+			"err", err,
+		)
+		return 0
+	}
+
+	return len
+}
+
+// AppendValues utlizes bbolts NextSequence functionality to add ordered values
+// after generating the next autoincrementing key for each
+func (s *bboltKeyValueStore) AppendValues(values ...[]byte) error {
+	if s == nil || s.db == nil {
+		return fmt.Errorf("unable to append values into uninitialized bbolt db store")
+	}
+
+	return s.db.Update(func(tx *bbolt.Tx) error {
+		b := tx.Bucket([]byte(s.bucketName))
+		if b == nil {
+			return NewNoBucketError(s.bucketName)
+		}
+
+		for _, value := range values {
+			key, err := b.NextSequence()
+			if err != nil {
+				return fmt.Errorf("generating key: %w", err)
+			}
+
+			b.Put(byteKeyFromUint64(key), value)
+			if err != nil {
```

Comment:
Is this supposed to be `if err := b.Put ... err != nil`?

Reply:
yes! great catch thank you